File: 2026POTD/FEBRUARY/03-02-2026/3637.cpp
class Solution {
public:
    bool isTrionic(vector<int>& nums) {
        int n = nums.size();
        int i = 0;

        // Phase 1: strictly increasing
        while(i + 1 < n && nums[i] < nums[i + 1])
            i++;

        if(i == 0 || i == n - 1) return false;

        // Phase 2: strictly decreasing
        int p = i;
        while(i + 1 < n && nums[i] > nums[i + 1])
            i++;

        if(i == p || i == n - 1) return false;

        // Phase 3: strictly increasing
        int q = i;
        while(i + 1 < n && nums[i] < nums[i + 1])
            i++;

        return i == n - 1;
    }
};

File: 2026POTD/FEBRUARY/03-02-2026/3637.java
class Solution {
    public boolean isTrionic(int[] nums) {
        int n = nums.length;
        boolean first = false;
        boolean second = false;
        boolean third = false;

        int i = 1;
        while(i < n){
            if(nums[i] > nums[i-1]){
                i++;
                first = true;
            }else{
                break;
            }
        }
        while(i < n){
            if(nums[i] < nums[i-1]){
                i++;
                second = true;
            }else{
                break;
            }
        }
        while(i < n){
            if(nums[i] > nums[i-1]){
                i++;
                third = true;
            }else{
                break;
            }
        }
        
        if(i == n){
            if(first && second && third){
                return true;
            }
        }
        return false;

        
    }
}